app/assets/javascripts/dradis/plugins/csv/upload.js

```diff
@@ -34,5 +34,57 @@ document.addEventListener('turbolinks:load', function() {
       $nodeSelect.closest('tr').find('[data-behavior=na-field-label]').removeClass('d-none');
       $nodeSelect.closest('tr').find('[data-behavior=default-field-label]').addClass('d-none');
     });
+
+    $('[data-behavior~=mapping-form]').on('ajax:before', function() {
+      $('#console').empty();
+      $('#result').show();
+    });
+
+    $('[data-behavior~=mapping-form]').on('ajax:complete', function() {
+      ConsoleUpdater.parsing = true;
+      setTimeout(ConsoleUpdater.updateConsole, 1000);
+    });
+
+    $('[data-behavior~=mapping-form]').submit(function() {
+      var valid = validateUniqueId() && validateNodeSelected();
+
+      if (!valid) {
+        $(this).find('input[type="submit"]').attr('disabled', false).val('Import CSV');
+
+        $('[data-behavior~=view-content]').animate({
+          scrollTop: $('[data-behavior~=mapping-validation-messages]').scrollTop()
+        });
+      }
+
+      return valid;
+    });
+
+    function validateUniqueId() {
```

Comment:
Do we really need this whole block of code instead of just adding `required: true` for the `f.radio_button` call?

Reply:
:man_facepalming: I forgot about `required: true`.